upload/src/addons/SV/WarningImprovements/XF/Entity/Warning.php

```diff
@@ -12,16 +12,67 @@ class Warning extends XFCP_Warning
 {
     public function canView(&$error = null)
     {
-        return (parent::canView($error) && $this->canViewIssuer());
+        $visitor = \XF::visitor();
+
+        if (!$visitor->user_id)
+        {
+            return false;
+        }
+
+        if ($visitor->user_id === $this->user_id && $this->app()->options()->sv_view_own_warnings)
+        {
+            $handler = $this->getHandler();
+            return $handler ? $handler->canViewContent($this) : false;
```

Comment:
This prevents a user from seeing a warning for content which has been deleted or is inaccessible. A return true works.

Reply:
Done